### PR TITLE
Persistent filters on refresh, added new cookies

### PR DESCRIPTION
**Pull request by javuto**
Implementation for #241, this pull request adds a new javascript module, ```filter.js```, that persist the values of the filters module in a set of cookies.

### CODE REVIEW

File: src/static/js/fb-ctf.js [hunk 1]
```diff
@@ -20,9 +21,11 @@ var widgetsList = [
   'Filter',
   'Game Clock'
 ];
-
 Widget.rememberWidgets(widgetsList);
 
+// This will be populated in loadSavedFilterModule
+var filterList = {};
```
Comment (gsingh93) [inline src/static/js/fb-ctf.js]: Since this is a list, let's initialize with square brackets for clarity.

File: src/static/js/filter.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
+var $ = require('jquery');
+
+function isFilterSet(filterName) {
+  return document.cookie.search(filterName) >= 0;
+}
+
+module.exports = {
+  getFilterName: function(filterSelector, filters) {
+    for (var key in filters) {
```
Comment (gsingh93) [inline src/static/js/filter.js]: Don't use `for..in` with JavaScript arrays: https://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-a-bad-idea

File: src/static/js/filter.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    var expires = "expires=" + d.toUTCString();
+    document.cookie = filterName + "=" + filterValue + "; " + expires;
+  },
+
+  getFilterState: function(filterName) {
+    var name = filterName + "=";
+    var ca = document.cookie.split(';');
+    for (var i = 0; i < ca.length; i++) {
+      var c = ca[i];
+      while (c.charAt(0) == ' ') c = c.substring(1);
```
Comment (gsingh93) [inline src/static/js/filter.js]: Always use curly braces

File: src/static/js/filter.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      if (c.indexOf(name) === 0) return c.substring(name.length, c.length);
+    }
+    return '';
+  },
+
+  rememberFilters: function(filters) {
+    for (var key in filters) {
```
Comment (gsingh93) [inline src/static/js/filter.js]: Don't use `for..in`

File: src/static/js/filter.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+      if (isFilterSet(key)) {
+        var ftype;
```
Comment (gsingh93) [inline src/static/js/filter.js]: Here and everywhere else for new code, let's use `let` instead of `var`.

File: src/static/js/filter.js (hunk 2 above)
Comment (javuto) [inline src/static/js/filter.js]: I could use ```Object.keys``` and then iterate through those...

**Pull request merged**